src/cmpStorage.ts

```diff
@@ -0,0 +1,153 @@
+import { storage, getCookie, setCookie } from '@guardian/libs';
+import { onConsent } from './onConsent';
+
+/**
+ * List of permitted use-cases for cookies/storage:
+ *
+ * - `Targeted advertising`: if the user can be targeted for personalised advertising according to the active consent framework
+ * - 'Targeted marketing': if the user can be targeted for personalised marketing, e.g. article count
+ * - `Essential`: if essential cookies/storage can be used according to the active consent framework
+ *
+ */
+export const UseCaseOptions = [
+	"Targeted advertising",
+	"Targeted marketing",
+	"Essential"
+] as const;
+export type UseCases = typeof UseCaseOptions[number];
+
+export const hasConsentForUseCase = async (useCase: UseCases): Promise<boolean> =>
+{
+	const consentState = await onConsent();
+
+	/*console.log(`consentState.tcfv2?.consents['1']: ${consentState.tcfv2?.consents['1']}`);
+	console.log(`consentState.tcfv2?.consents['2']: ${consentState.tcfv2?.consents['2']}`);
+	console.log(`consentState.tcfv2?.consents['3']: ${consentState.tcfv2?.consents['3']}`);
+	console.log(`consentState.tcfv2?.consents['4']: ${consentState.tcfv2?.consents['4']}`);
+	console.log(`consentState.tcfv2?.consents['5']: ${consentState.tcfv2?.consents['5']}`);
+	console.log(`consentState.tcfv2?.consents['6']: ${consentState.tcfv2?.consents['6']}`);
+	console.log(`consentState.tcfv2?.consents['7']: ${consentState.tcfv2?.consents['7']}`);
+	console.log(`consentState.tcfv2?.consents['8']: ${consentState.tcfv2?.consents['8']}`);
+	console.log(`consentState.tcfv2?.consents['9']: ${consentState.tcfv2?.consents['9']}`);
+	console.log(`consentState.tcfv2?.consents['10']: ${consentState.tcfv2?.consents['10']}`);
+	console.log(`consentState.tcfv2?.consents['11']: ${consentState.tcfv2?.consents['11']}`);
+	console.log(`consentState.canTarget: ${consentState.canTarget}`);
+	*/
+
+	switch(useCase) {
```

Comment:
I think there might be one other use-case for permissions 1 and 2 - "Limited data advertising" - I'm going to ask Data Privacy if Article Count can switch to use [1 & 2] rather than [1, 3 & 7].